### ISSUE TITLE
Crashes on Android 10

### ISSUE THREAD

**Issue by ellyasafaa** [onaio/rdt-standard#950]
It happened on the freshly installed APK on 3 devices: 1 Samsung A50s and 2 Samsung A21.

How I get the crashes:
1. Installed the APK
2. Waiting to sync
3. Click one of the patients
4. Click RDT Form
5. Click Locate Me
6. Click Continue until the page of Which RDT?
7. Click the barcode scanning option
8. Scanning the barcode (After getting caught, sometimes the screen appeared twice after being scanned)
9. Status: OK page appeared, click continue
10. White page appeared with RDT written on the top of the page

Other crashes:
1. It successfully scanned until the capturing image page appeared
2. After capturing the image, white page appeared with RDT written on the top of the page

The video is here:
https://drive.google.com/file/d/105Qzc-QyJzYY7u9t64PivP49yD0lCgFP/view?usp=sharing

@yussin-hidayat @vincent-karuri @mwkhan1983 @syedowaisali90

**Comment by syedowaisali90** [comment]
Hi @ellyasafaa I have reproduced this if the returned `GTIN/Product Id` will empty then the information will not be displayed. could you give me some sample barcodes, so that I can run some more tests.

cc: @vincent-karuri @mwkhan1983

**Comment by ellyasafaa** [comment]
@syedowaisali90 Hi Owais, thank you. I have shared with you the folders with some barcodes in it.

**Comment by syedowaisali90** [comment]
@ellyasafaa App working fine on those barcodes which you have been shared with me, could you confirm it was reproducing on your side or it happened only once. Moreover, scanning is performing on a separate APK, so maybe this was happened due to the opening camera.

**Comment by yussin-hidayat** [comment]
Maybe these logs can help. This happened after first time login and sync:
[error.log](https://github.com/onaio/rdt-standard/files/6282524/error.log)

And this after re-login:
[error_2.log](https://github.com/onaio/rdt-standard/files/6282525/error_2.log)

**Comment by syedowaisali90** [comment]
@yussin-hidayat could you confirm that was happening every fresh installed? and if so, this was happening only specific device or any other device.

**Comment by anuraj-shankar** [comment]
Hi All, thanks for the intense effort on this Ellya, Yussin, Owais and others. @syedowaisali90 do you have a way of logging the exact execution point where the crash is occurring? Are you able to test different phones to replicate this?

**Comment by yussin-hidayat** [comment]
@syedowaisali90 I have tried 5 times fresh installed, always crashes after I clicked "new patient" button. I use emulator Pixel 2 API 29.

**Comment by mwkhan1983** [comment]
@yussin-hidayat you may be trying that on Master branch and Android 10 compatibility fix is not in Master yet. This was actually the original issue which was resolved. @syedowaisali90 created a branch for this and shared release with Ellya. Owais, if you can share that branch with Yussin, he can check this thing there.

**Comment by syedowaisali90** [comment]
@yussin-hidayat you can check out this branch https://github.com/onaio/rdt-standard/tree/integrate-optimized-native-forms-branch it has Android 10 fixes.

**Comment by ellyasafaa** [comment]
Hi, tried it on devices Redmi 5 Plus Android 7, Oppo A5s Android 8, Samsung Galaxy A21s Android 10, Samsung Galaxy A50s Android 10, and the error happened on all devices if we pressed the back button on the scanning page, that will impact to crash and error of unable to click the page: https://drive.google.com/file/d/11zNpfFooynvpsbYHPe5ii2LTznMc_xXR/view?usp=sharing (on Oppo A5s Android 8)

But, specifically for Android 10, it also happened after scanning or capturing the image: https://drive.google.com/file/d/11hmuZx9zW6wHAG7GibGXyHf9dui9gyNp/view?usp=sharing (this video showed when the error happened after scanning, on Samsung Galaxy A21s Android 10)

**Comment by mwkhan1983** [comment]
@ellyasafaa following is the separate issue and is noted here #916. and Owais has recently fixed it.

> Hi, tried it on devices Redmi 5 Plus Android 7, Oppo A5s Android 8, Samsung Galaxy A21s Android 10, Samsung Galaxy A50s Android 10, and the error happened on all devices if we pressed the back button on the scanning page, that will impact to crash and error of unable to click the page: https://drive.google.com/file/d/11zNpfFooynvpsbYHPe5ii2LTznMc_xXR/view?usp=sharing (on Oppo A5s Android 8)

**Comment by ellyasafaa** [comment]
@mwkhan1983 thank you. Has this been merged in the latest apps you sent me?
